### ISSUE TITLE
Tabs are encoded as &#x9;

### ISSUE THREAD

**Issue by gkjohnson** [gkjohnson/xacro-parser#20]
Possibly this is a result of using XMLSerializer? It's an issue in tests here:

https://github.com/gkjohnson/xacro-parser/blob/82c758d941f996a024b5ec5dc2445768035c7012/test/URDFGeneration.test.js#L93

**Comment by gkjohnson** [comment]
It seems to be that the `XMLSerializer` converts tabs to encoded `&#x9;` when they're present in attributes. However the parser strips them out of attributes and converts them to spaces (unless they're encoded) so it's odd they're included in the result at all. Tabs are retained if they're in text content, though, so it's possible that the tabs are present in a macro which is unpacked for an attribute.

```js
const elem = document.createElement('test');
elem.setAttribute('attr', '\t \t');
elem.textContent = '\t \t';
const result = new XMLSerializer().serializeToString(elem);
// <test xmlns="http://www.w3.org/1999/xhtml" attr="&#9; &#9;">	 	</test>
```

**Comment by gkjohnson** [comment]
Some research should be done to see what the python serializer and parser does.

**Comment by gkjohnson** [comment]
Python seems to freely write tabs as `\t` in attributes but also converts them to spaces when parsing a document.

**Comment by gkjohnson** [comment]
For the valkyrie file the tabs are present in the .`/test/data/val_description/model/robots/common/xacro/arm/mass_parameters.xacro` in the `ShoulderYawLinkComPosition`, `ElbowPitchLinkComPosition`, and `ForearmYawLinkComPosition` fields. It's unclear why those values are not getting converted to spaces when the xml is parsed.

**Comment by gkjohnson** [comment]
This was an issue with the JSDom parser not behaving like the browsers. Fixed by updating it in 7a0cb38.
